### PR TITLE
[GEOT-6739] GroupCandidateSelectionProcess should return feature with null attribute

### PR DESCRIPTION
**Pull request by taba90**
Jira ticket https://osgeo-org.atlassian.net/browse/GEOT-6739

- This pr fix a problem with GroupCandidateSelection process, making it returning a feature aslso if in a defined group the operationValue is always NULL for all the members.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/GroupCandidateSelectionProcess.java [hunk 1]
```diff
@@ -343,23 +342,26 @@ else if (groupingValues != null
         }
 
         private Feature updateBestFeature(Feature best, Feature f) {
+            Comparable bestValue = getComparableFromEvaluation(best);
+            Comparable value = getComparableFromEvaluation(f);
+            if (bestValue == null && value == null) return best;
```
Comment (aaime) [inline modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/GroupCandidateSelectionProcess.java:347]: So, 

```
if (value == null) return best;
else if (bestValue == null) return f;
```

File: modules/unsupported/process-feature/src/test/resources/org/geotools/process/vector/test-data/featuresToGroupWithGroupAllNull.properties [hunk 2]
```diff
@@ -0,0 +1,13 @@
+_=the_geom:MultiPolygon:srid=4326,cat:java.lang.Long,group:String,group2:String,numericVal:java.lang.Integer
+features.1=MULTIPOLYGON(((2 2,2 3,3 3,3 2,2 2)), ((3 9,3 12,5 12,5 9,3 9)))|10|group1|secondGroup1|null
```
Comment (aaime) [inline modules/unsupported/process-feature/src/test/resources/org/geotools/process/vector/test-data/featuresToGroupWithGroupAllNull.properties:2]: Is it just me, or there is no group with all values at null? I can only see groups with a mix of nulls and non nulls (which needs to be tested for sure, but the title talks about features with null attribute).

File: modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/GroupCandidateSelectionProcess.java (hunk 1 above)
Comment (taba90) [reply modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/GroupCandidateSelectionProcess.java]: indeed fixed
